src/main/java/com/google/devtools/build/lib/exec/SpawnStrategyRegistry.java

```diff
@@ -595,6 +631,61 @@ public String toString() {
     }
   }
 
+  private static class StrategyPlatformFilter {
+    private final ImmutableListMultimap<Label, SpawnStrategy> platformToStrategies;
+
+    private StrategyPlatformFilter(
+      ImmutableListMultimap<Label, SpawnStrategy> platformToStrategies) {
+      this.platformToStrategies = platformToStrategies;
+    }
+
+    /**
+     * Gets strategies for the given spawn that are allowed by the execution platform.
+     * @param spawn Spawn to pick strategies for. Must have an execution platform.
+     * @param candidateStrategies Strategies ordered by priority to pick from. The contents of this
+     *                            list vary depending on the context but are always the result of
+     *                            an initial strategy selection pass.
+     *                            e.g. {@link SpawnStrategyRegistry#getStrategies(Spawn, EventHandler)},
+     *                            {@link SpawnStrategyRegistry#getDynamicSpawnActionContexts(Spawn, DynamicMode)}
+     *                            and {@link SpawnStrategyRegistry#getRemoteLocalFallbackStrategy(Spawn)}.
+     * @return A subset of {@code candidateStrategies} that are allowed by the spawn's execution
+     *         platform or all if no restrictions are in place.
+     *         Order from {@code candidateStrategies} is preserved.
+     */
+    public <T extends SpawnStrategy> List<T> getStrategies(
+        Spawn spawn, List<T> candidateStrategies) {
```

Comment:
```suggestion
        Spawn spawn, Collection<T> candidateStrategies) {
```
so that an `ImmutableCollection can be passed in directly without the copy-on-write adapter.

Reply:
Switching to `Collection` here is an easy enough change.

Removing the copy-on-write adapter is a lot more "viral" of a change, spreading all the way to `SpawnStrategyRegistry.getStrategies(Spawn, EventHandler)`. A refactor such as that is deserving of it's own PR.